frontend/lib/providers/filtered_test_execution_ids.dart

```diff
@@ -1,25 +1,42 @@
 import 'package:dartx/dartx.dart';
 import 'package:riverpod_annotation/riverpod_annotation.dart';
 
+import '../models/filters.dart';
+import '../models/test_execution.dart';
 import 'artefact_builds.dart';
-import 'test_execution_filters.dart';
 
 part 'filtered_test_execution_ids.g.dart';
 
 @riverpod
 Set<int> filteredTestExecutionIds(
   FilteredTestExecutionIdsRef ref,
   int artefactId,
+  Uri pageUri,
 ) {
   final builds = ref.watch(artefactBuildsProvider(artefactId)).requireValue;
   final testExecutions = [
     for (final build in builds)
       for (final testExecution in build.testExecutions) testExecution,
   ];
-  final filters = ref.watch(testExecutionFiltersProvider(artefactId));
+  final filters =
+      emptyTestExecutionFilters.copyWithQueryParams(pageUri.queryParametersAll);
+  final searchValue = pageUri.queryParameters['q'] ?? '';
```

Comment:
Sorry for a dumb question, but is it always defined as a value for the `q` key?

Reply:
Yes search queries are typically always have the key `q` (e.g. `https://www.google.com/search?q=asdf`). In our case, when the user applies the filters, I make sure to add the search query under this key too. Of course it could be the case that there is no search query, in that case `?? ''` kicks in and gives it a default value.